### ISSUE TITLE
[QUESTION] How to import config file?

### ISSUE THREAD

**Issue by Larvotto** [Lissy93/dashy#772]
### Question

I've exported a config file via Update Configuration > Download as File. That gives me a yaml file in the following format (snippet copied from VSCode):

![image](https://user-images.githubusercontent.com/73801020/176344902-23da094f-3c0f-4cb6-84e5-d256b3554358.png)

However, I can't see how to import this back into Dashy. Update Configuration > Edit Config uses the Tree or Code formats, neither of which are consistent with the format of the exported yaml file: Tree uses collapsible sections, and Code uses " and {} symbols that are stipped out of the export - so I can't simply paste the exported code straight back in.

I'm probably missing something basic, so would be very grateful for any assistance. 

### Category

Configuration

### Please tick the boxes

- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy (check the first two digits of the version number)
- [X] You've checked that this [question hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)

**Comment by Lissy93** [comment]
Heya @Larvotto 

You can use something like [YAML to JSON](https://jsonformatter.org/yaml-to-json), then in the config editor, select Code mode, and paste the contents in. 

JSON and YAML have exactly the same structure, just YAML uses indentation instead of curly braces, commas and quotes.
The mismatch is a bit inconsistent, I agree. But it's because yaml = nicer for humans, json = nicer for computers :)

**Comment by Larvotto** [comment]
Excellent, thanks. I knew it would be something basic!

BTW, the exported config - either to clipboard or a file - includes a rogue space at the very beginning, i.e., before "appConfig", which causes an error if not removed. No biggie to fix, but thought you might want to know.

BTW#2 - Dashy is excellent! Thanks so much for all your hard work.

**Comment by Lissy93** [comment]
Fix for that whitespace will be include in #799 :)
ETA mid-August.

**Comment by yannduran** [comment]
I've just had the same problem with the rogue space in the exported config file in v2.1.1

```
Error, unable to validate 'conf.yml' 
⚠️ Error on line 8, column 0: end of the stream or a document separator is expected
  6 |       path: https://github.com/ ...
  7 |     - title: Documentation
  8 |       path: https://dashy.to/docs
  9 | appConfig:
------^
 10 |   theme: nord-frost
 11 |   layout: auto
```

When I opened the file in VS Code, I found that the `pageInfo` section header was incorrectly indented by one space.

```
 pageInfo:
  title: Dashy
  description: Welcome to your new dashboard!
  navLinks:
    - title: GitHub
      path: https://github.com/Lissy93/dashy
    - title: Documentation
      path: https://dashy.to/docs
appConfig:
```

**Comment by johncarterofmars** [comment]
I tried importing the converted json. The file imports in, and I wait for the rebuild. However, the only thing that will change is the theme. None of the individual boxes import, and the default items are still there. This is after saving to disk (not locally). 
Any suggestions?

**Comment by neohtpc** [comment]
Same issue here, no data after such import

**Comment by dabulls45** [comment]
I had to restore today. And this was the only thread I found about the issue. So what I did:

//Go to:

https://jsonformatter.org/yaml-to-json

//Paste in saved config text, convert it, save to file

// Dashy config editor, select Code mode, and paste the contents in

Click on Wrench/"Update config", Edit Config, switch from Tree to Code by clicking on Tree, at the top. Replace the JSON text with the converted text/JSON from saved file. Save changes. You will see the changes immediately after.

I had trouble finding to Code Editor.

**Comment by xdpirate** [comment]
I landed here after updating my ancient Dashy docker container and trying to restore my config. My data imported fine once I ran it through a YAML > JSON tool, but an import button that takes the file exported is sorely missing.

> Are you serious? You have the option to download a backup file of the configuration, but no way to import it? are you kidding me?

This an incredibly rude comment, but the point is valid—it's a very odd design choice to have an export function that exports to a different markup language than the language expected at import time.
